server/src/com/thoughtworks/go/config/GoConfigPluginService.java

```diff
@@ -23,8 +27,18 @@
  */
 @Component
 public class GoConfigPluginService {
+
+    private final XmlPartialConfigProvider embeddedXmlPlugin;
+
+    @Autowired public GoConfigPluginService(ConfigCache configCache,ConfigElementImplementationRegistry configElementImplementationRegistry,
+                                 MetricsProbeService metricsProbeService)
+    {
+        MagicalGoConfigXmlLoader loader = new MagicalGoConfigXmlLoader(configCache, configElementImplementationRegistry, metricsProbeService);
+        embeddedXmlPlugin = new XmlPartialConfigProvider(loader);
```

Comment:
I see that `XmlPartialConfigProvider` is not really a Go *plugin*. Do you plan to introduce an extension point and move this to a plugin in one of your later PRs?